src/main/scala/com/yotpo/metorikku/output/writers/file/FileOutputWriter.scala

```diff
@@ -73,9 +73,7 @@ class FileOutputWriter(props: Map[String, Object], outputFile: Option[File]) ext
         catalog.tableExists(tableName) match {
           // Quick overwrite (using alter table + refresh instead of drop + write + refresh)
           case true => {
-            writer.save()
-            log.info(s"Overwriting external table $tableName to new path $filePath")
-            ss.sql(s"ALTER TABLE $tableName SET LOCATION '$filePath'")
+            writer.saveAsTable(tableName)
```

Comment:
Why?

Reply:
@lyogev "saveAsTable" already updates the location. and it also updates the schema, when "save" does not.

Reply:
There's a reason I did it like I did (it's possible it's missing something).
Check out saveAsTable source code:
```
      case (true, SaveMode.Overwrite) =>
...
        // Drop the existing table
        catalog.dropTable(tableIdentWithDB, ignoreIfNotExists = true, purge = false)
        createTable(tableIdentWithDB)
        // Refresh the cache of the table in the catalog.
        catalog.refreshTable(tableIdentWithDB)
```
Note that it drops the table and then it creates it (this includes writing the actual files).
This will create a point in time when someone can access hive and get nothing....